### ISSUE TITLE
Missing plugin crashes with IndexError: list index out of range

### ISSUE THREAD

**Issue by berkes** [beetbox/beets#3062]
### Problem

Running the `missing` plugin with no options, or any option but the `-a` will crash. Only the `-a` allows it to run.

Running this command in verbose (`-vv`) mode:

```sh
$ beet -vv missing
```

Led to this problem:

```
user configuration: /home/xxx/.config/beets/config.yaml
data directory: /home/xxx/.config/beets
plugin paths: 
Sending event: pluginload
library database: /home/xxx/.local/share/beets/musiclibrary.blb
library directory: /media/xxx/ExtraDrive1/Muziek
Sending event: library_opened
Traceback (most recent call last):
  File "/home/xxx/.local/bin/beet", line 11, in <module>
    sys.exit(main())
  File "/home/xxx/.local/lib/python2.7/site-packages/beets/ui/__init__.py", line 1256, in main
    _raw_main(args)
  File "/home/xxx/.local/lib/python2.7/site-packages/beets/ui/__init__.py", line 1243, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/home/xxx/.local/lib/python2.7/site-packages/beetsplug/missing.py", line 126, in _miss
    helper(lib, decargs(args))
  File "/home/xxx/.local/lib/python2.7/site-packages/beetsplug/missing.py", line 155, in _missing_tracks
    for item in self._missing(album):
  File "/home/xxx/.local/lib/python2.7/site-packages/beetsplug/missing.py", line 219, in _missing
    if len([i for i in album.items()]) < album.albumtotal:
  File "/home/xxx/.local/lib/python2.7/site-packages/beets/dbcore/db.py", line 360, in __getattr__
    return self[key]
  File "/home/xxx/.local/lib/python2.7/site-packages/beets/dbcore/db.py", line 252, in __getitem__
    return getters[key](self)
  File "/home/xxx/.local/lib/python2.7/site-packages/beets/library.py", line 1093, in _albumtotal
    return self.items()[0].tracktotal
  File "/home/xxx/.local/lib/python2.7/site-packages/beets/dbcore/db.py", line 651, in __getitem__
    return self._objects[n]
IndexError: list index out of range

```
I'm not sure if any music files might cause this behaviour.

### Setup

* OS: Ubuntu 18.04.1 LTS
* Python version: 2.7.15rc1
* beets version: 1.4.7
* Turning off plugins made problem go away (yes/no): No, the 'missing' plugin is required to run the missing command.

My configuration (output of `beet config`) is:

```yaml
directory: /media/xxx/ExtraDrive1/Muziek/
asciify_paths: yes
library: ~/.local/share/beets/musiclibrary.blb

plugins: duplicates missing
missing:
    count: no
    album: no
    total: no
duplicates:
    count: no
    full: no
    format: ''
    keys: []
    move: ''
    tag: ''
    path: no
    copy: ''
    tiebreak: {}
    album: no
    strict: no
    checksum: ''
    merge: no
    delete: no

```

**Comment by sampsyo** [comment]
Huh! This is strange—it looks like there's an empty album (an Album structure with no associated Item entries) in your library. Can you confirm that? You might try `beet ls -a '$albumtotal'`, for example, and see where that crashes.

This shouldn't happen: when the last item is removed from an album, it should disappear. Any guesses as to how this might have come about?

**Comment by berkes** [comment]
It looks like I indeed have albums without tracks. No idea how that came to be. My beets database is several years old, and I have moved it around a lot and have tried some plugins on it, in the past.

Your suggestion with `ls -a -f '$albumtotal` crashed immediately; with the same backtrace. It did not show if and where possible titles were having no items associated, though. 

The database, however, did give me some albums without items:

```SQL
sqlite> select count(*) from albums left join items on items.album_id = albums.id WHERE items.id IS NULL;
741
```

I'm not 100% sure of the database structure, but I guess deleting them from the SQLite should work. 

So I backupped the database, then ran:
```SQL
delete from albums where id IN (select albums.id from albums left join items on items.album_id = albums.id WHERE items.id IS NULL);
```

Now both ` beet -vv ls -a -f'$album - tracks: $albumtotal'` and `beet missing` work!

I'll close this for now. The SQL might help a future googler. But, dear future Googler, ensure you backup first; good chance you'll break the data :ghost:
